### PR TITLE
Added Image Metadata documentation

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

**Comment by mfojtik** [comment]
@adellape @bparees @soltysh PTAL

File: image_writers_guide/metadata.adoc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+images other than specifying extra environment variables. There is upstream
+work currently being done to fix this situation and we should switch to that in
+the implementation in the future.
+
+This document only defines the metadata needed by the current set of use cases.
+Additional metadata and/or use cases may be added in the future.
+
+== How to define metadata for Docker images
+
+Right know, the only method is it use the `ENV` instruction in the Docker file.
```
Comment (soltysh) [inline image_writers_guide/metadata.adoc]: in the Dockerfile.

File: image_writers_guide/metadata.adoc [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+The environment variables are also available in the Docker image JSON
+representation, where the platform can see them and use them.
+
+== Supported metadata variables
+
+=== IMAGE_TAGS
+
+This metadata variable contains a list of tags. The tags are the way to
+categorize Docker images into broad areas of functionality. Tags helps UI and
```
Comment (soltysh) [inline image_writers_guide/metadata.adoc]: Tags help~~s~~

File: image_writers_guide/metadata.adoc [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+Example format:
+
+----
+IMAGE_TAGS=database,mysql
+----
+
+=== IMAGE_WANTS
+
+Specifies a list of tags that the generation tools and UI might use for
+suggesting to provide if you don't have the Docker images with given tags
```
Comment (soltysh) [inline image_writers_guide/metadata.adoc]: tools and UI might ~~use for~~ suggest~~ing~~ to provide

File: image_writers_guide/metadata.adoc [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+Example format:
+
+----
+IMAGE_TAGS=database,mysql
+----
+
+=== IMAGE_WANTS
+
+Specifies a list of tags that the generation tools and UI might use for
```
Comment (soltysh) [inline image_writers_guide/metadata.adoc]: Specifies a list of tags ~~that~~ the generation tools

File: image_writers_guide/metadata.adoc [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+don't have the Docker image with 'redis' tag, then UI might suggest you to add
+this image into your application.
+
+The format of the values is the same as for `IMAGE_TAGS`.
+
+=== IMAGE_DESCRIPTION
+
+This variable is used to give the image consumers more detailed description
+about the service or functionality given image provides.
+The UI can then use this description together with the image name to give users
```
Comment (soltysh) [inline image_writers_guide/metadata.adoc]: he UI can ~~then~~ use this

File: image_writers_guide/metadata.adoc [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+more informations about the image.
```
Comment (soltysh) [inline image_writers_guide/metadata.adoc]: information~~s~~ 

Information is uncountable, so there's not s in the end.

File: image_writers_guide/metadata.adoc [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+Example:
+
+----
+IMAGE_EXPOSE_SERVICES="3128/tcp:mysql,8080:http"
+----
+
+=== IMAGE_NON_SCALABLE
+
+An image might use this variable to suggest that it does not support scaling.
```
Comment (soltysh) [inline image_writers_guide/metadata.adoc]: to suggest ~~that~~ it does

File: image_writers_guide/metadata.adoc [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+The UI will then communicate this to consumers of that image.
+Being not-scalable basically means that the value of 'replicas' should initially
```
Comment (soltysh) [inline image_writers_guide/metadata.adoc]: basically means ~~that~~ the value

**Comment by mfojtik** [comment]
@soltysh thanks a lot! should be fixed now

**Comment by bparees** [comment]
maybe we should link this from the main image writer guidelines doc?

**Comment by adellape** [comment]
@bparees That makes sense to me.

Is there an existing guideline in `guidelines.adoc` where some quick info on image metadata and a link to this new topic would fit well? Maybe "Set Environment Variables"? Or does this warrant its own blurb under the "OpenShift-Specific Guidelines" section there?

**Comment by bparees** [comment]
@adellape the existing guidelines doc is pretty lengthy as it is, i'd rather start breaking it up by putting this in its own doc.

**Comment by bparees** [comment]
@adellape sorry I may have misunderstood your question.  I took a quick skim of the existing guidelines doc and unfortunately I don't see a natural fit for this info in any of the existing sections.

**Comment by adellape** [comment]
@bparees OK, yea I just meant finding a place in `guidelines.adoc` to link to this new topic.

Then I guess a new, short "Set Image Metadata" guideline (that is basically just a link to `metadata.adoc`) would be appropriate.

**Comment by bparees** [comment]
@adellape +1

**Comment by CowboysFan** [comment]
@bparees @adellape These topics for the most part should be able to stand on their own. If this topic is completely dependent on something else taking place, then I don't think it should be on its own. Also, the metadata.adoc topic will shrink some because we'll wrap those Env Variables in a table, so it may not be as long as we think. Are these current constraints/assumptions specific to OpenShift? Are these optional or a required component?

Maybe it would make sense to have a topic in the beginning that describes what components an OpenShift image contains? Probably in a table with this information: 

OpenShift Image Components
  STI
  Services for Inter-image Communication
  .......
  Image Metadata

But I see that the STI Requirements topic has a section titled "Required Image Contents". Could the metadata topic be part of that? In general I think it may be good to provide customers the components of what we think an OpenShift image consist of, or should consist of, and include the metadata topic along with other requirements with links to their respective topics. Hope that makes sense.

**Comment by bparees** [comment]
they are specific to openshift and they are recommendations, not requirements.   if image authors provide this info, we can be smarter about how we consume the image.  therefore it's similar to the other recommendations we make about how to create images for openshift.

the recommendations are not specific to STI, they are specific to openshift, so putting them in the STI section doesn't make sense.

We definitely don't want a topic that describes "what an openshift image contains" because there's no such thing as an openshift image.  We consume all images, we just have recommendations for how to design your image for a better experience if you are in a position to do so.

**Comment by CowboysFan** [comment]
@bparees Awesome, thanks for that explanation and I see the challenge now. I guess I'm just not sure of it being its own topic, but maybe we'll get a better picture as we get more content. I'm okay to go with it being a separate topic for the moment, and since I'll be going through our content to determine the best structure of our topics, we can revisit this at a later time.

**Comment by mfojtik** [comment]
@CowboysFan @bparees so are you guys OK to have this as a separate topic under image writers guide?

**Comment by bparees** [comment]
@mfojtik I like it better on a separate page mainly just because it's so lengthy compared to the other sections of the image writer's guide.  Maybe that's not a good reason, but it just seems more in depth/detailed than those other sections so this doesn't feel like it belongs there.  But if it's simpler for now to shove it into that page, i'm ok with it.  I'm also ok with leaving it as is but adding a link from the image writer's guide page.

**Comment by adellape** [comment]
I'm going to go ahead and merge this for now, and I'll add a link to it in the Guidelines topic in a follow-up PR.

**Comment by mfojtik** [comment]
@adellape thanks!

**Pull request merged**